Makefile

```diff
@@ -566,52 +803,86 @@ clean: .cleaninst.LINUX.clean .cleaninst.LINUX32.clean .cleaninst.WIN32.clean .c
+.stage.%.mkspiffs .stage.%.mklittlefs .stage.%.esptool: CC=$(call host,$@)-gcc
+.stage.%.mkspiffs .stage.%.mklittlefs .stage.%.esptool: CXX=$(call host,$@)-g++
+.stage.%.mkspiffs .stage.%.mklittlefs .stage.%.esptool: STRIP=$(call host,$@)-strip
+
+# arm builds using clang, not gcc.
+# same as .stage.%.strip - simply stamp the target, never change it
+.stage.MACOSARM.mkspiffs .stage.MACOSARM.mklittlefs .stage.MACOSARM.esptool: CC=$(call host,$@)-cc
+.stage.MACOSARM.mkspiffs .stage.MACOSARM.mklittlefs .stage.MACOSARM.esptool: CXX=$(call host,$@)-c++
+.stage.MACOSARM.mkspiffs .stage.MACOSARM.mklittlefs .stage.MACOSARM.esptool: STRIP=touch
+
 .stage.%.done: .stage.%.package .stage.%.mkspiffs .stage.%.esptool .stage.%.mklittlefs
 	echo STAGE: $@
 	echo Done building $(call arch,$@)
+	touch $@
 
-# Only the native version has to be done to install libs to GIT
-install: .stage.LINUX.install
-.stage.LINUX.install:
-	echo STAGE: $@
-	rm -rf $(ARDUINO)
-	git clone https://github.com/$(GHUSER)/Arduino $(ARDUINO)
-	(cd $(ARDUINO) && git checkout $(INSTALLBRANCH) && git submodule init && git submodule update)
-	echo "-------- Building installable hal"
-	rm -rf arena/hal-install; mkdir -p arena/hal-install
-	cd arena/hal-install; $(call setenv,$@); $(REPODIR)/lx106-hal/configure --prefix=$(ARDUINO)/tools/sdk/libc --libdir=$(ARDUINO)/tools/sdk/lib --host=xtensa-lx106-elf $$(echo $(call configure,$@) | sed 's/--host=.*\s//' | sed 's/--prefix=.*\s//' | sed 's/--disable-libstdcxx-verbose//' )
-	cd arena/hal-install; $(call setenv,$@); $(MAKE) ; $(MAKE) install
-	echo "-------- Copying GCC libs"
-	#cp $(call install,$@)/lib/gcc/xtensa-lx106-elf/*/libgcc.a  $(ARDUINO)/tools/sdk/lib/.
-	cp $(call install,$@)/xtensa-lx106-elf/lib/libstdc++-exc.a $(ARDUINO)/tools/sdk/lib/.
-	cp $(call install,$@)/xtensa-lx106-elf/lib/libstdc++.a     $(ARDUINO)/tools/sdk/lib/.
+.stage.LINUX.arduino-checkout:
+	echo "-------- Preparing Arduino repo at $(ARDUINO)"
+	test -d $(ARDUINO) || git clone https://github.com/$(GHUSER)/Arduino $(ARDUINO)
+	(cd $(ARDUINO) \
+		&& git clean -f -d \
+		&& git fetch origin $(INSTALLBRANCH) \
+		&& git checkout $(INSTALLBRANCH) \
+		&& git submodule init \
+		&& git submodule update)
+
+.stage.LINUX.arduino-toolchain:
+	echo "-------- Copying GCC and LIBSTDC++ libs"
+	cp -vu $(call install,$@)/$(TARGET_ARCH)/lib/libstdc++-exc.a $(ARDUINO)/tools/sdk/lib/.
+	cp -vu $(call install,$@)/$(TARGET_ARCH)/lib/libstdc++.a     $(ARDUINO)/tools/sdk/lib/.
 	echo "-------- Copying toolchain directory"
-	rm -rf $(ARDUINO)/tools/sdk/xtensa-lx106-elf
-	cp -a $(call install,$@)/xtensa-lx106-elf $(ARDUINO)/tools/sdk/xtensa-lx106-elf
+	rm -rf $(ARDUINO)/tools/sdk/$(TARGET_ARCH)
+	cp -va $(call install,$@)/$(TARGET_ARCH) $(ARDUINO)/tools/sdk/$(TARGET_ARCH)
+
+.stage.LINUX.arduino-hal:
+	echo "-------- Copying HAL lib"
+	cp -vu $(call install,$@)/$(TARGET_ARCH)/lib/libhal.a $(ARDUINO)/tools/sdk/lib/.
```

Comment:
any reason for .a copies distributed with release or git (aka tools/get.py)?
the ones in the tools/sdk/lib actually take priority based on search path injected in both PIO and platform.txt

afaik, all three are linking just fine from {toolchain}/xtensa-lx106-elf/lib already
both pio & ide